e2e/api.py

```diff
@@ -27,10 +25,9 @@ def get_api_url(self, path):
         Args:
             path (str): API endpoint path.
         """
-        if not self.append_slash:
-            path.strip('/')
-        elif not path.endswith('/'):
-            path = path.lstrip('/') + '/'
+        path = path.strip('/')
+        if self.append_slash:
+            path += '/'
 
         return f"{self.api_url_root.strip('/')}/{path}"  # type: ignore
```

Comment:
I would also update f-string to `urljoin` to create the URL it will look better:

```
return urljoin(f"{self.api_url_root}/", path)
```